File: 05-threads/05-threads1.c
//
// basic program to create a thread
// Author: David Kerschbaumer
//

#include <stdio.h>
#include <pthread.h>
#include <unistd.h>

void* runFunction()
{
  printf("Hello, I'm another thread\n");
  return NULL;
}


int main()
{
  printf("Hello, I'm the first-Thread\n");
  pthread_t thread_id;
  pthread_create(&thread_id, NULL, runFunction, NULL);
  pthread_join(thread_id, NULL);
  return 0;
}

File: 05-threads/05-threads1.py
#
# basic program to create a thread
# Author: David Kerschbaumer
#

import threading
import time

def my_function():
  time.sleep(2)
  print("Hello from thread!")

my_thread = threading.Thread(target=my_function) # Create a thread
my_thread.start() # Start the thread
my_thread.join() # Wait for the thread to finish (optional)

print("Main thread have finished.")
